### ISSUE TITLE
Error when trying to set path parameter in front of static query

### ISSUE THREAD

**Issue by NoahKamara** [joshuawright11/papyrus#47]
When trying to set a path parameter that sits in front of a static query like `bar/:baz?query=1` papyrus fails to find the path parameter:

```swift
PapyrusError(
  message: "Tried to set path parameter `baz` but did not find `:baz` in path `bar/:baz?query=1`.", 
  request: nil, 
  response: nil
)
```

Test Case:
```swift
func testPathWithStaticQuery() {
        var req = RequestBuilder(baseURL: "foo/", method: "GET", path: "bar/:baz?query=1")
        req.addParameter("baz", value: "value")
        
        XCTAssertNoThrow(try req.fullURL())
        XCTAssertEqual(try req.fullURL().absoluteString, "foo/bar/value/?query=1")
    }
```

I'd be willing to work on this, since i need it for a project

**Comment by joshuawright11** [comment]
Thanks for the report @NoahKamara ! Want to create a PR from your fixed repo?

**Comment by joshuawright11** [comment]
Fixed in #48
